workload: support import parallelism for multi-tenant cluster

stevendanna opened the pull request:
All nodes now produce entries in the sql_instances table, so we are able to use that table to get a node count even when running against a secondary tenant.

I've kept the original query in place to support older clusters.

Fixes #78968

Release note: None

stevendanna committed:
workload: support import parallelism for multi-tenant cluster

All nodes now produce entries in the sql_instances table, so we are
able to use that table to get a node count even when running against a
secondary tenant.

I've kept the original query in place to support older clusters.

Fixes #78968

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96904)
<!-- Reviewable:end -->

adityamaru reviewed (approved):
LGTM, i don't know if we have tests in workloadccl but in case you want to add a small unit test.

stevendanna committed:
workloadccl: add missing log scope calls

Release note: None

stevendanna committed:
workloadccl: enable tenant testing

I've run these under stress for a few hundred iterations and haven't
encountered a problem.

Release note: none

adityamaru reviewed (approved):
Thanks!

stevendanna commented:
bors r=adityamaru
